ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSsystem.cpp

```diff
@@ -51,20 +53,26 @@ clock_t lasttime;
 using std::string;
 using std::stringstream;
 
-#include "Platforms/Win32/WINDOWSmain.h"
 IDirectSoundBuffer* primaryBuffer;
 
 vector<SoundResource*> sound_resources(0);
 
 namespace enigma {
 
+extern HWND hWnd;
+
 void eos_callback(void* soundID, unsigned src) {
   get_sound(snd, (ptrdiff_t)soundID, );
   snd->playing = false;
   snd->idle = true;
 }
 
 int audiosystem_initialize() {
+  if (get_window_handle() == NULL) {
+    MessageBox(NULL, "Window handle is NULL.", "Error", MB_OK);
```

Comment:
I... I can't raise much of a stink about this, Robert, because the function *will* exist, but why not just use `show_error` for UX consistency?

Reply:
Well, I did not add this one, but I am responsible for other places that MessageBox is called. Specifically for the `SetCooperativeLevel` and such.
https://github.com/enigma-dev/enigma-dev/blob/a66f43f896ab5307bcba7941b5a7db65f710e712/ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSsystem.cpp#L86

I am not sure what my thinking was, it may have just been easier to call MessageBox than figure out how to include `show_error` for me at the time. I was also probably reading DirectSound and DirectX tutorials to make these systems, so I didn't think about it.

You could argue that there's no need to call `show_error` here, since it will call `MessageBox` anyway. I don't really have an opinion on this though. I think my reasoning could have been that I did not want to show an option to continue, since this is basically a fatal error.